ioctl/cmd/ins/insregister.go

```diff
@@ -0,0 +1,352 @@
+          "name": "commitment",
+          "type": "bytes32"
+        }
+      ],
+      "name": "commit",
+      "outputs": [],
+      "stateMutability": "nonpayable",
+      "type": "function"
+    },
+		{
+      "inputs": [
+        {
+          "internalType": "string",
+          "name": "name",
+          "type": "string"
+        },
+        {
+          "internalType": "uint256",
+          "name": "duration",
+          "type": "uint256"
+        }
+      ],
+      "name": "rentPrice",
+      "outputs": [
+        {
+          "components": [
+            {
+              "internalType": "uint256",
+              "name": "base",
+              "type": "uint256"
+            },
+            {
+              "internalType": "uint256",
+              "name": "premium",
+              "type": "uint256"
+            }
+          ],
+          "internalType": "struct IPriceOracle.Price",
+          "name": "price",
+          "type": "tuple"
+        }
+      ],
+      "stateMutability": "view",
+      "type": "function"
+    },
+		{
+      "inputs": [
+        {
+          "internalType": "string",
+          "name": "name",
+          "type": "string"
+        },
+        {
+          "internalType": "address",
+          "name": "owner",
+          "type": "address"
+        },
+        {
+          "internalType": "uint256",
+          "name": "duration",
+          "type": "uint256"
+        },
+        {
+          "internalType": "bytes32",
+          "name": "secret",
+          "type": "bytes32"
+        },
+        {
+          "internalType": "address",
+          "name": "resolver",
+          "type": "address"
+        },
+        {
+          "internalType": "bytes[]",
+          "name": "data",
+          "type": "bytes[]"
+        },
+        {
+          "internalType": "bool",
+          "name": "reverseRecord",
+          "type": "bool"
+        },
+        {
+          "internalType": "uint16",
+          "name": "ownerControlledFuses",
+          "type": "uint16"
+        }
+      ],
+      "name": "register",
+      "outputs": [],
+      "stateMutability": "payable",
+      "type": "function"
+    }
+	]`
+)
+
+// _insRegisterCmd represents the action hash command
+var _insRegisterCmd = &cobra.Command{
+	Use:   config.TranslateInLang(_registerUsages, config.UILanguage),
+	Short: config.TranslateInLang(_registerShorts, config.UILanguage),
+	Args:  cobra.MinimumNArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		err := register(args)
+		return output.PrintError(err)
+	},
+}
+
+func init() {
+	_insRegisterCmd.Flags().StringVarP(
+		&controller, "controller", "c",
+		"0x8aA6acF9BFeEE0243578305706766065180E68d4",
+		config.TranslateInLang(_registerControllerUsages, config.UILanguage),
+	)
+	_insRegisterCmd.Flags().StringVarP(
+		&resolver, "resolver", "r",
+		"0x41B9132D4661E016A09a61B314a1DFc0038CE3e8",
+		config.TranslateInLang(_registerResolverUsages, config.UILanguage),
+	)
```

Comment:
looks like `controller, resolver` are constant address, so we don't need to set as flag here? can directly define as `const` or assign value in `var ()`

Reply:
The `controller` and `resolver` default value is mainnet contract address, the testnet address is different from mainnet.

Reply:
i see